lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb

```diff
@@ -2,99 +2,88 @@
+      it('товар не опубликован') { expect(@cs_eti_table_products.public_state(@product)).to eq :unpublished }
     end
 
     context 'когда товар с рубрикой' do
       before(:all) do
         @name = Faker::Number.number(5)
-        @cs_eti_page.add_product
-        @cs_eti_page.set_name(@name)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.set_rubric(CONFIG['eti']['rubric'])
-        @cs_eti_page.wait_until { @cs_eti_page.first_product_status_element.attribute('title') == 'Опубликованные' }
-        @cs_eti_page.refresh
-        @cs_eti_page.search_product(@name)
+        @cs_eti_table_products.add_product(name: @name, rubric: CONFIG['eti']['rubric'])
+        @cs_eti_header.search_product(@name, exact: true)
+        @product = @cs_eti_table_products.product(name: @name)
       end
 
-      it 'введенное имя отображается' do
-        expect(@cs_eti_page.product_name?(@name)).to be true
-      end
+      after(:all) { @cs_eti_table_products.delete_product(@product) }
 
-      it 'рубрика привязана' do
-        expect(@cs_eti_page.product_rubric_tree(@name)).to include CONFIG['eti']['rubric']
-      end
-
-      it 'товар опубликован' do
-        expect(@cs_eti_page.product_published?(@name)).to be true
-      end
-
-      after(:all) { @cs_eti_page.delete_product(@name) }
+      it('введенное имя отображается') { expect(@cs_eti_table_products.name(@product)).to eq @name }
+      it('рубрика привязана') { expect(@cs_eti_table_products.rubric(@product)).to include CONFIG['eti']['rubric'] }
+      it('товар опубликован') { expect(@cs_eti_table_products.public_state(@product)).to eq :published }
     end
 
     context 'когда копируем товар' do
       before(:all) do
-        @product = {
-          name: Faker::Name.title,
+        @fields = {
+          name: Faker::Job.title[0..74],
           rubric: CONFIG['eti']['rubric'],
-          exists: CONFIG['eti']['exists']['in stock'],
-          short_description: CONFIG['product_creation']['short_description']['valid'],
-          description: 'description',
-          price_from_to: {from: Faker::Number.number(3), to: Faker::Number.number(4)},
-          wholesale_price: {wholesale_price: Faker::Number.number(2), wholesale_number: Faker::Number.number(2)}
-        }
-
-        @portal_traits = {
-          trait_1: CONFIG['eti']['portal_traits']['trait_value_1'],
-          trait_2: CONFIG['eti']['portal_traits']['trait_value_2']
+          exists: :available,
+          announce: Faker::Hobbit.quote[0..254],
+          description: Faker::Hobbit.quote,
+          price: {
+            type: :range,
+            price: Faker::Number.number(3),
+            price_max: Faker::Number.number(4)
+          },
+          wholesale_price: {
+            price: Faker::Number.number(2),
+            min_qty: Faker::Number.number(2)
+          },
+          traits: {
+              CONFIG['eti']['portal_traits']['trait_1'] => CONFIG['eti']['portal_traits']['trait_value_1'],
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.